### PR TITLE
Fix asserts and related stubs.

### PR DESCRIPTION
**Pull request by jmdobry**

### CODE REVIEW

File: bigquery/test/queries.test.js [hunk 1]
```diff
@@ -108,115 +111,107 @@ describe('bigquery:query', function () {
...
-      assert.ifError(callback.firstCall.args[0]);
-      assert(example.mocks.job.getQueryResults.called);
-      assert(console.log.calledWith(
-        'AsyncQuery: polled job %s; got %d rows!',
-        example.jobId,
-        example.mocks.natality.length
-      ));
+      assert.equal(example.mocks.job.getMetadata.calledOnce, true);
+      assert.deepEqual(example.mocks.job.getMetadata.firstCall.args.slice(0, -1), []);
+      assert.equal(example.mocks.job.getQueryResults.calledOnce, true);
```
Comment (ace-n) [inline bigquery/test/queries.test.js]: Do we need to check the `args` of this?

File: storage/test/transfer.test.js [hunk 2]
```diff
@@ -77,50 +77,46 @@ describe('storage:transfer', function () {
...
         sample.program.createTransferJob(options, callback);
 
-        assert(callback.calledTwice, 'callback called twice');
-        assert.equal(callback.secondCall.args.length, 2, 'callback received 2 arguments');
-        assert.ifError(callback.secondCall.args[0], 'callback did not receive error');
-        assert.strictEqual(callback.secondCall.args[1], sample.mocks.transferJob, 'callback received transfer job');
-        assert.equal(sample.mocks.storagetransfer.transferJobs.create.secondCall.args[0].resource.description, description, 'description was set');
-        assert(console.log.calledWith('Created transfer job: %s', sample.mocks.transferJob.name));
+        assert.equal(sample.mocks.storagetransfer.transferJobs.create.calledTwice, true);
+        assert.equal(sample.mocks.storagetransfer.transferJobs.create.secondCall.args[0].resource.description, description);
```
Comment (ace-n) [inline storage/test/transfer.test.js]: `assert.deepEqual`?

File: storage/test/transfer.test.js [hunk 3]
```diff
@@ -77,50 +77,46 @@ describe('storage:transfer', function () {
...
         sample.program.createTransferJob(options, callback);
 
-        assert(callback.calledOnce, 'callback called once');
-        assert.equal(callback.firstCall.args.length, 2, 'callback received 2 arguments');
-        assert.ifError(callback.firstCall.args[0], 'callback did not receive error');
-        assert.strictEqual(callback.firstCall.args[1], sample.mocks.transferJob, 'callback received transfer job');
-        assert.equal(sample.mocks.storagetransfer.transferJobs.create.firstCall.args[0].resource.description, undefined, 'description was not set');
-        assert(console.log.calledWith('Created transfer job: %s', sample.mocks.transferJob.name));
+        assert.equal(sample.mocks.storagetransfer.transferJobs.create.calledOnce, true);
+        assert.equal(sample.mocks.storagetransfer.transferJobs.create.firstCall.args[0].resource.description, undefined);
```
Comment (ace-n) [inline storage/test/transfer.test.js]: `assert.deepEqual`?

Comment (ace-n) [inline storage/test/transfer.test.js]: (never mind)

File: storage/test/transfer.test.js (hunk 2 above)
Comment (ace-n) [inline storage/test/transfer.test.js]: (never mind)

File: storage/test/transfer.test.js [hunk 4]
```diff
@@ -179,31 +176,69 @@ describe('storage:transfer', function () {
...
 
         sample.program.updateTransferJob(options, callback);
 
-        assert(callback.calledTwice, 'callback called twice');
-        assert.equal(callback.secondCall.args.length, 2, 'callback received 2 arguments');
-        assert.ifError(callback.secondCall.args[0], 'callback did not receive error');
-        assert.strictEqual(callback.secondCall.args[1], sample.mocks.transferJob, 'callback received transfer job');
+        assert.equal(sample.mocks.storagetransfer.transferJobs.patch.calledTwice, true);
+        assert.deepEqual(sample.mocks.storagetransfer.transferJobs.patch.secondCall.args.slice(0, -1), [{
+          auth: {},
```
Comment (ace-n) [inline storage/test/transfer.test.js]: Can we put this in a `var` easily?

File: storage/test/transfer.test.js [hunk 5]
```diff
@@ -305,53 +340,66 @@ describe('storage:transfer', function () {
...
+          name: 'transferOperations',
+          auth: {},
+          filter: JSON.stringify({ project_id: process.env.GCLOUD_PROJECT, job_names: [jobName] })
+        }]);
+        assert.equal(callback.calledTwice, true);
+        assert.deepEqual(callback.secondCall.args, [null, [sample.mocks.transferOperation]]);
+        assert.equal(console.log.calledTwice, true);
+        assert.deepEqual(console.log.secondCall.args, ['Found %d operations!', 1]);
+
+        sample.mocks.storagetransfer.transferOperations.list.yields(null, {});
```
Comment (ace-n) [inline storage/test/transfer.test.js]: Maybe add a comment as to what each of these tests does differently?

**Comment by ace-n** [comment]
LGTM assuming all (relevant) tests pass on Travis - just a few minor nits.

File: bigquery/test/queries.test.js (hunk 1 above)
Comment (jmdobry) [inline bigquery/test/queries.test.js]: Done.

File: storage/test/transfer.test.js (hunk 5 above)
Comment (jmdobry) [inline storage/test/transfer.test.js]: Done.

**Comment by jmdobry** [comment]
The one failing travis test is unrelated. Apparently one of the eslint dependencies committed a `const` and Node.js 0.12.x throws a strange error when it sees the `const`.

**Pull request merged**